src/transformers/models/omdet_turbo/modeling_omdet_turbo.py

```diff
@@ -0,0 +1,2034 @@
+
+        context_layer = self.out_proj(context_layer)
+
+        outputs = (context_layer, attention_probs) if output_attentions else (context_layer,)
+
+        return outputs
+
+
+class OmDetTurboEncoderLayer(nn.Module):
+    def __init__(self, config: OmDetTurboConfig):
+        super().__init__()
+        self.normalize_before = config.normalize_before
+        self.self_attn = OmDetTurboMultiheadAttention(
+            config.encoder_hidden_dim, config.num_attention_heads, config.dropout
+        )
+        self.self_attn_layer_norm = nn.LayerNorm(config.encoder_hidden_dim, eps=config.layer_norm_eps)
+        self.dropout = config.dropout
+        self.activation_fn = ACT2FN[config.ffn_encoder_activation]
+        self.activation_dropout = config.activation_dropout
+        self.fc1 = nn.Linear(config.encoder_hidden_dim, config.encoder_dim_feedforward)
+        self.fc2 = nn.Linear(config.encoder_dim_feedforward, config.encoder_hidden_dim)
+        self.final_layer_norm = nn.LayerNorm(config.encoder_hidden_dim, eps=config.layer_norm_eps)
+
+    @staticmethod
+    def with_pos_embed(tensor, pos_embed):
+        return tensor if pos_embed is None else tensor + pos_embed
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: torch.Tensor,
+        position_embeddings: torch.Tensor = None,
+        output_attentions: bool = False,
+    ):
+        """
+        Args:
+            hidden_states (`torch.FloatTensor`): input to the layer of shape `(batch, seq_len, embed_dim)`
+            attention_mask (`torch.FloatTensor`): attention mask of size
+                `(batch, 1, target_len, source_len)` where padding elements are indicated by very large negative
+                values.
+            position_embeddings (`torch.FloatTensor`, *optional*):
+                Object queries (also called content embeddings), to be added to the hidden states.
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+        """
+        residual = hidden_states
+        if self.normalize_before:
+            hidden_states = self.self_attn_layer_norm(hidden_states)
+        query = key = self.with_pos_embed(hidden_states, position_embeddings)
+
+        # TODO address this problem
+        # Transpose to have the equivalent of batch_first=False. batch_first should have been set to True in the
+        # original model implementation, but it is not the case. This is a workaround to have consistent results
+        # with the original model, but means that the model does not support batch inference.
+        hidden_states = self.self_attn(
+            queries=query.transpose(1, 0),
+            keys=key.transpose(1, 0),
+            values=hidden_states.transpose(1, 0),
+            attention_mask=attention_mask,
```

Comment:
Sorry about that. Since the correct implementation seems to be with batch_first=True as confirmed by the authors, we don't need to transpose at all now.